client/src/Component/SignUp/formValidation.js

```diff
@@ -0,0 +1,41 @@
+import validator from 'validator';
+
+const checkValidity = (value, rule, myFormState) => {
+  let valid = true;
+  if (rule.required) {
+    valid = value.trim() === '' ? false : true && valid;
+  }
+  if (rule.validEmail) {
+    valid = validator.isEmail(value) && valid;
+  }
+  if (rule.minLength) {
+    valid = value.trim().length >= '4' && valid;
+  }
+  if (rule.matchWithPassword) {
+    valid = myFormState.userPassword.value === value && valid;
+  }
+
+  return valid;
+};
+const completeFormValidation = (myFormState) => {
+  const formData = {
```

Comment:
you can make this generic by mapping out all the keys using object.keys().map, so that you can use with all the forms.

Reply:
Done